cmd/kops/integration_test.go

```diff
@@ -71,81 +71,92 @@ func TestHighAvailabilityGCE(t *testing.T) {
 
 // TestComplex runs the test on a more complex configuration, intended to hit more of the edge cases
 func TestComplex(t *testing.T) {
-	runTestAWS(t, "complex.example.com", "complex", "v1alpha2", false, 1)
+	runTestAWS(t, "complex.example.com", "complex", "v1alpha2", false, 1, true)
 }
 
 // TestMinimalCloudformation runs the test on a minimum configuration, similar to kops create cluster minimal.example.com --zones us-west-1a
 func TestMinimalCloudformation(t *testing.T) {
-	runTestCloudformation(t, "minimal.example.com", "minimal-cloudformation", "v1alpha2", false)
+	runTestCloudformation(t, "minimal.example.com", "minimal-cloudformation", "v1alpha2", false, nil)
+}
+
+// TestExistingIAMCloudformation runs the test with existing IAM instance profiles, similar to kops create cluster minimal.example.com --zones us-west-1a
+func TestExistingIAMCloudformation(t *testing.T) {
+	lifecycleOverrides := []string{"IAMRole=Ignore", "IAMRolePolicy=Ignore", "IAMInstanceProfileRole=Ignore"}
```

Comment:
Would `LifecycleExistsAndWarnIfChanges` allow us to use managed policies as opposed to inline policies? The role has identical permissions but the IAM resources will not be in the state that the awstasks are expecting. In our case we use `iam:AttachRolePolicy` rather than `iam:PutRolePolicy`, so the `IAMRolePolicy` awstask would see that the inline policy is missing.